src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -1233,11 +1233,18 @@ rpmostree_compose_builtin_install (int             argc,
     return FALSE;
   g_assert (self); /* Pacify static analysis */
   gboolean changed;
-  if (!impl_install_tree (self, &changed, cancellable, error))
-    {
-      self->failed = TRUE;
-      return FALSE;
-    }
+  /* Need to handle both GError and C++ exceptions here */
+  try {
+    if (!impl_install_tree (self, &changed, cancellable, error))
+      {
+        self->failed = TRUE;
+        return FALSE;
+      }
+  } catch (std::exception &e) {
+    self->failed = TRUE;
+    rpm_ostree_tree_compose_context_free (self); self = NULL;
```

Comment:
(I did briefly look at converting the struct to C++ RAII but a lot higher risk/churn for lower reward versus eventually hopefully rewriting this code in Rust)